test/unit/step.spec.js

```diff
@@ -91,21 +98,22 @@ describe('Tour | Step', () => {
           action: instance.next
         }
       ],
-      popperOptions: {
-        modifiers: [{ name: 'offset', options: { offset: [0, 0] } }]
+      floatingUIOptions: {
+        middleware: [stepWithoutNameWithoutIdOffsetMiddleware]
       }
     });
 
-    const beforeShowPromise = new Promise((resolve) => {
-      return setTimeout(() => resolve('beforeShowPromise worked!'), 1000);
+    const beforeShowPromise = () => new Promise((resolve) => {
```

Comment:
Changed the nesting, we don't access this promise directly, we're looking at the output of the show() method to get the promise.